parseutil/parseutil.go

```diff
@@ -106,19 +106,28 @@ func ParseDurationSecond(in interface{}) (time.Duration, error) {
 			return dur, nil
 		}
 		var err error
-		// Look for a suffix otherwise its a plain second value
-		if strings.HasSuffix(inp, "s") || strings.HasSuffix(inp, "m") || strings.HasSuffix(inp, "h") || strings.HasSuffix(inp, "ms") {
+
+		suffix := inp[len(inp)-1:]
+		scale := time.Second
+
+		switch suffix {
+		case "s", "m", "h":
+			// Use ParseDuration if string has a supported suffix. Note that "ms" is covered by "s".
 			dur, err = time.ParseDuration(inp)
 			if err != nil {
 				return dur, err
 			}
-		} else {
-			// Plain integer
-			secs, err := strconv.ParseInt(inp, 10, 64)
+		case "d":
+			// Parse days (which aren't supported by the stdlib) by stripping the suffix and adjusting the scale.
+			scale = 24 * time.Hour
+			inp = inp[:len(inp)-1]
+			fallthrough
+		default:
+			val, err := strconv.ParseFloat(inp, 64)
```

Comment:
Want to call out that changing this to `strconv.ParseFloat` means that we now accept fractional values so it's a behavioral change.

Reply:
I only today learned that the suffix variants already supported fractional:

```
➜  ~ vault token create -ttl=5.5h
Key                  Value
---                  -----
token                s.rfMyxR65n7GZARo9A1wvdwFk
token_accessor       0kjRm43pa71ODDeY1KeY0KMd
token_duration       5h30m
token_renewable      true
token_policies       ["root"]
identity_policies    []
policies             ["root"]
```

So supporting fractional for days seemed logical, but you're right that is does change the parsing for the no-suffix number.

Reply:
Actually, ParseFloat introduces a bunch of unwanted edgecases. I'll revert to ParseInt and we'll only allow integral days.